File: SDKs/googleapis.com/alertcenter/v1beta1/java/lib/src/main/java/openapisdk/models/shared/BatchUndeleteAlertsRequest.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * BatchUndeleteAlertsRequest
 * A request to perform batch undelete on alerts.
**/
public class BatchUndeleteAlertsRequest {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("alertId")
    public String[] alertId;
    public BatchUndeleteAlertsRequest withAlertId(String[] alertId) {
        this.alertId = alertId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("customerId")
    public String customerId;
    public BatchUndeleteAlertsRequest withCustomerId(String customerId) {
        this.customerId = customerId;
        return this;
    }
}

File: SDKs/googleapis.com/alertcenter/v1beta1/python/src/sdk/models/shared/batchundeletealertsrequest.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class BatchUndeleteAlertsRequest:
    r"""BatchUndeleteAlertsRequest
    A request to perform batch undelete on alerts.
    """
    
    alert_id: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('alertId') }})
    customer_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('customerId') }})
    
